### PR TITLE
fix(asgi.App): correctly pass params when using `orjson.dumps`

### PR DESCRIPTION
**Pull request by CaselIT**
# Summary of Changes

Replace this text with a high-level summary of the changes included in this PR.

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

#2100 

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/latest/quickstart.html#creating-news-fragments) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: docs/_newsfragments/2100.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix an issue when using orjson as json serializer with asyncio.
```
Comment (vytas7) [inline docs/_newsfragments/2100.bugfix.rst:1]: This newsfragment could be improved: we shouldn't use the imperative mood here (as in docstrings), but tell a story what has been done. Also, we should capitalize JSON, and that `asyncio` sounds a bit vague, maybe we could be more explicit that it is `falcon.asgi.App` we're talking about.

File: falcon/media/json.py [hunk 2]
```diff
@@ -190,7 +190,9 @@ def _serialize_s(self, media, content_type=None) -> bytes:
     async def _serialize_async_s(self, media, content_type) -> bytes:
         return self._dumps(media).encode()
 
-    def _serialize_b(self, media, content_type) -> bytes:
+    # NOTE(kgriffs): Make content_type a kwarg to support the
+    #   Request.render_body() shortcut optimization.
+    def _serialize_b(self, media, content_type=None) -> bytes:
         return self._dumps(media)
 
     async def _serialize_async_b(self, media, content_type) -> bytes:
```
Comment (vytas7) [inline falcon/media/json.py:198]: Do we have a test exercising `_serialize_async_b()` in an App (as opposed to just isolated unit tests)?

**Review by vytas7 — CHANGES_REQUESTED** [review]
LGTM except the newsfragment :+1:

File: falcon/media/json.py (hunk 2 above)
Comment (CaselIT) [reply falcon/media/json.py:198]: yes, the test I added. We were missing it before: `test_full_app`

without the fix it fails with the error in question

Comment (vytas7) [reply falcon/media/json.py:198]: I meant `_serialize_async_()`, not `_serialize_b()`. We haven't touched that method in this PR, hence asking if there are tests to ensure no bugs are lurking there either.

File: docs/_newsfragments/2100.bugfix.rst (hunk 1 above)
Comment (CaselIT) [reply docs/_newsfragments/2100.bugfix.rst]: updated. hopefully it's better than before

File: falcon/media/json.py (hunk 2 above)
Comment (CaselIT) [reply falcon/media/json.py:198]: ok, let's try subclassing the json handlers in the same test. that should prevent the shortcut

Comment (CaselIT) [reply falcon/media/json.py:198]: done

**Review by vytas7 — APPROVED** [review]
LGTM now :shipit:

**Comment by CaselIT** [comment]
thanks for the update on the news fragment

**Pull request merged**